aws-janitor/resources/set.go

```diff
@@ -89,11 +94,16 @@ func (s *Set) Save(sess *session.Session, p *s3path.Path) error {
 
 	svc := s3.New(sess, aws.NewConfig().WithRegion(p.Region))
 
+	// Field Tagging must be encoded as URL Query parameters, (For example, "Key1=Value1")
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#PutObjectInput
+	stateTag := fmt.Sprintf("%s=true", JanitorStateTag)
+
 	_, err = svc.PutObject(&s3.PutObjectInput{
 		Bucket:       aws.String(p.Bucket),
 		Key:          aws.String(p.Key),
 		Body:         bytes.NewReader(b),
 		CacheControl: aws.String("max-age=0"),
+		Tagging:      &stateTag,
```

Comment:
For now, it's really hard to check whether the object is a janitor state object or not since its key is derived from the "path" flag 
https://github.com/youyangl/boskos/blob/8ed406269010654ac4a50e70e62abf17864a7ef5/cmd/aws-janitor/main.go#L45. which can be customized by user.
One possible solution is that we can set the "Tagging" field when we store object in bucket.
However, one issue I'm concern is how to rollout this change. Before this change, janitor state objects don't have any tag, which means that when deleting expired buckets, those objects might be deleted together.
From my perspective, it should not be a big issue:
1. Only resources in GlobalTypeList https://github.com/youyangl/boskos/blob/8ed406269010654ac4a50e70e62abf17864a7ef5/aws-janitor/resources/list.go#L107 will be influenced by this issue.
2. Even though the janitor state object got deleted unintentionally, the next time when we run the janitor, the new object will be generated with tag. The "firstseen" map will be regenerated using the timestamp of the new run. I don't think it will cause too much quota issue.

Reply:
Taking the feature is behind a feature flag, we probably can just be explicit in flag docs about need to tag state buckets with do not delete tags. Any heuristics that would work in common case feels complex and fragile, and there will be likely edge cases we don't envision anyway that would cause surprising behavior.

Reply:
That make sense, we can check the bucket tag every time we set the "path" flag in aws-janitor/main.go, only when the bucket is not managed by janitor, we can do the deletion.